planet/cli/data.py

```diff
@@ -23,83 +45,66 @@ def parse_item_types(ctx, param, value: str) -> List[str]:
 
 def parse_filter(ctx, param, value: str) -> dict:
     """Turn filter JSON into a dict."""
-    return json.loads(value)
-
-
-@click.group()
-@click.pass_context
-def data(ctx):
-    """A group of commands for interacting with the Data API."""
+    # read filter using raw json
+    if value.startswith('{'):
+        try:
+            json_value = json.loads(value)
+        except json.decoder.JSONDecodeError:
+            raise click.ClickException('File does not contain valid json.')
```

Comment:
@mkshah605 I recommend the following. It will cause click to print out a helpful error message about the problems with the input. I think it would be good if these callback functions only raise BadParameter.

```suggestion
        except json.decoder.JSONDecodeError as err:
            raise click.BadParameter(err, ctx=ctx, param=param)
```

Reply:
hmmm even with this, I'm seeing the error when I run the user-side code, but it's not registering when I run the test. The test is failing because it's expecting an error to be raised but that's not currently happening. Somehow it looks like the error is not registering?

Reply:
@mkshah605 correct! When there's a bad parameter, we don't want to propagate an exception. Instead we want to print a message and set an exit code of 2. And that's what click does when it encounters a BadParameter. See number 2 under https://click.palletsprojects.com/en/8.1.x/exceptions/?highlight=badparameter#where-are-errors-handled.

Reply:
Also, I strongly recommend passing the JSON decode error to BadParameter as I show above. The error message is very specific.